server/src/main/java/org/opensearch/common/blobstore/AsyncMultiStreamEncryptedBlobContainer.java

```diff
@@ -44,12 +48,16 @@ public void asyncBlobUpload(WriteContext writeContext, ActionListener<Void> comp
 
     @Override
     public void readBlobAsync(String blobName, ActionListener<ReadContext> listener) {
-        throw new UnsupportedOperationException();
-    }
-
-    @Override
-    public void asyncBlobDownload(String blobName, Path fileLocation, ThreadPool threadPool, ActionListener<String> completionListener) {
-        throw new UnsupportedOperationException();
+        try {
+            ActionListener<ReadContext> decryptingCompletionListener = ActionListener.map(
+                listener,
+                readContext -> new DecryptedReadContext<>(readContext, cryptoHandler, getEncryptedHeaderContentSupplier(blobName))
```

Comment:
Minor, but I'd construct the cryptoContext here and make the ctor of DecryptedReadContext just do field assignment with no logic. Something like:

```
readContext -> {
    final U context = cryptoHandler.loadEncryptionMetadata(getEncryptedHeaderContentSupplier(blobName));
    return new DecryptedReadContext<>(readContext, cryptoHandler, context);
}
```